staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go

```diff
@@ -212,55 +218,126 @@ func (o *AutoscaleOptions) Run() error {
 			return fmt.Errorf("cannot autoscale a %v: %v", mapping.GroupVersionKind.Kind, err)
 		}
 
-		hpa := o.createHorizontalPodAutoscaler(info.Name, mapping)
-
-		if err := o.Recorder.Record(hpa); err != nil {
-			klog.V(4).Infof("error recording current command: %v", err)
+		if err := o.handleHPACreation(info.Name, mapping); err != nil {
+			return err
 		}
 
-		if o.dryRunStrategy == cmdutil.DryRunClient {
-			count++
-
-			printer, err := o.ToPrinter("created")
-			if err != nil {
-				return err
-			}
-			return printer.PrintObj(hpa, o.Out)
-		}
+		count++
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	if count == 0 {
+		return fmt.Errorf("no objects passed to autoscale")
+	}
+	return nil
+}
 
-		if err := util.CreateOrUpdateAnnotation(o.createAnnotation, hpa, scheme.DefaultJSONEncoder()); err != nil {
+// handleHPACreation handles the creation of HPA objects for both v2 and v1 APIs.
+// If v2 API fails, try to create and handle HPA using v1 API
+func (o *AutoscaleOptions) handleHPACreation(name string, mapping *meta.RESTMapping) error {
+	hpaV2 := o.createHorizontalPodAutoscalerV2(name, mapping)
+	if err := o.handleHPA(hpaV2); err != nil {
+		klog.Warningf("error handling v2 HPA: %v, trying v1 HPA", err)
```

Comment:
Change that to `klog.V(1).Info(...` message, it's not worth the warning.

Reply:
done